### PR TITLE
Restructure src directory

### PR DESCRIPTION
**Pull request by 36degrees**
## Objectives
- Restructure the `src` directory to make the structure of Frontend clearer, now that we are not confined by needing to structure around multiple packages.
- Simplify imports within components
- Add tests to ensure that the `settings`, `helpers` and `tools` layers do not output CSS.

## Summary

- Move components into `./components`
- Move `/icons` into ./assets/image
- Move `globals/settings` into `./settings`
- Move `globals/tools` into `./tools`
- Split `globals/helpers` so that mixins are in `./tools`, and concrete classes are in `./utilities`
- Move `globals/core` into `./core`
- Move `globals/objects` into `./objects`
- Move `globals/overrides` into `./overides`
- Move `mq()` into `./vendor`
- Move `globals/font-face.scss` into `./helpers/font-faces.scss` and refactor as a mixin
- Move `globals/common.js` to `./common.js`
- Move polyfills into `./vendor`
- Add entry points to each layer e.g. `settings/_all.scss`
- Update documentation

I suggest reviewing commit-by-commit.

## Before
```
src
├── back-link
├── breadcrumbs
├── button
├── checkboxes
├── date-input
├── details
├── error-message
├── error-summary
├── fieldset
├── file-upload
├── footer
├── globals
│   ├── core
│   ├── helpers
│   ├── objects
│   ├── overrides
│   ├── polyfills
│   │   ├── Function
│   │   │   └── prototype
│   │   └── Object
│   ├── settings
│   └── tools
├── hint
├── icons
├── input
├── label
├── panel
├── phase-banner
├── radios
├── select
├── skip-link
├── table
├── tag
├── textarea
└── warning-text
```

## After
```
src
├── assets
│   └── images
├── components
│   ├── back-link
│   ├── breadcrumbs
│   ├── button
│   ├── checkboxes
│   ├── date-input
│   ├── details
│   ├── error-message
│   ├── error-summary
│   ├── fieldset
│   ├── file-upload
│   ├── footer
│   ├── hint
│   ├── input
│   ├── label
│   ├── panel
│   ├── phase-banner
│   ├── radios
│   ├── select
│   ├── skip-link
│   ├── table
│   ├── tag
│   ├── textarea
│   └── warning-text
├── core
├── helpers
├── objects
├── overrides
├── settings
├── tools
├── utilities
└── vendor
    └── polyfills
        ├── Function
        │   └── prototype
        └── Object
```
https://trello.com/c/NsVi35Un/934-5-restructure-govuk-frontend-for-ease-of-consumption-and-maintainability

### CODE REVIEW

**Comment by igloosi** [comment]
i ran the test, package build and dist build and all passes and expected files are there. 👍 
once thing to decide is the package name in package.json

**Comment by 36degrees** [comment]
> once thing to decide is the package name in package.json

That needs to be done before we can do a release, but doesn't tie directly to this work? Might be worth creating a new card for it.

**Comment by igloosi** [comment]
no, not directly tied to this work.
decide name and update main readme.

File: docs/coding-standards/css.md [hunk 1]
```diff
@@ -1,123 +1,61 @@
...
-
-* [Harry Roberts - More Transparent UI Code with Namespaces](https://csswizardry.com/2015/03/more-transparent-ui-code-with-namespaces/)
-
-# BEM
-
-BEM – meaning block, element, modifier – is a front-end naming methodology.
-
-BEM tells developers how classes relate to each other.
+GOV.UK Frontend uses the Block Element Modifier (BEM) methodology when naming
+CSS classes. This is designed to help developers how the different classes
```
Comment (hannalaakso) [inline docs/coding-standards/css.md]: Missing a word, should probably say "help developers understand"

**Comment by nickcolley** [comment]
This looks awesome, do we want `common.js` at the root?

**Comment by 36degrees** [comment]
> This looks awesome, do we want common.js at the root? I was thinking potentially it makes sense in /utilities/all.js (since the entries are all) this allows us to break the file up without making a breaking change in the future?
> 
> Edit: I'm in two minds if it should actually be in 'tools' or 'utilities', so feel free to disregard this comment.

Yeah, we originally said it was going to go in utilities but it just felt super strange, especially when I was trying to describe the structure in `src/README.md`. I figured putting it in the root for now makes sense, we might end up e.g. creating a `common` folder and having a file for each function?

File: docs/coding-standards/css.md (hunk 1 above)
Comment (36degrees) [reply docs/coding-standards/css.md]: Good spot! 🦅

**Pull request merged**